### ISSUE TITLE
Variables names are not allowed to start with keywords

### ISSUE THREAD

**Issue by akshaymankar** [dhall-lang/dhall-haskell#1583]
Here is a simple example showing the unexpected error:

```
$ dhall <<< 'let mergeFoo = 99 in mergeFoo'
dhall:
Error: Invalid input

(stdin):1:27:
  |
1 | let mergeFoo = 99 in mergeFoo
  |                           ^^
unexpected "Fo"
expecting whitespace
```

**Comment by sjakobi** [comment]
The situation is similar with other keywords like `let` or `toMap`:

```
⊢ let letx = 1 in letx

Error: Invalid input

(stdin):1:20:
  |
1 | let letx = 1 in letx
  |                    ^
unexpected 'x'
expecting whitespace

⊢ let toMapX = 1 in toMapX

Error: Invalid input

(stdin):1:24:
  |
1 | let toMapX = 1 in toMapX
  |                        ^
unexpected 'X'
expecting whitespace
```

I'm not sure how difficult it would be to fix this.

**Comment by Gabriel439** [comment]
Fix is up here: #1584
